cloudferrylib/os/actions/check_openstack_apis.py

```diff
@@ -19,11 +19,14 @@
 LOG = utl.get_log(__name__)
 
 
-def check(os_api_call, os_api_type, position):
+def check(os_api_call, os_api_type, position, arg=None):
     try:
         LOG.info("Checking %s APIs availability on %s.",
                  os_api_type, position.upper())
-        os_api_call()
+        if arg is not None:
+            os_api_call(arg)
+        else:
+            os_api_call()
     except Exception as e:
```

Comment:
Let's also limit the exceptions caught here to all base exceptions of openstack clients, e.g.:
```
from neutronclient import exceptions as neutron_excs
from glanceclient import exceptions as glance_excs
...
try:
    ...
except (neutron_excs.BaseException, glance_excs.BaseException, ...) as e:
    ...
```